src/scripts/generate_covid_case_datamart.py

```diff
@@ -0,0 +1,131 @@
+            i.  If present an additional infection has been identified resulting in
+                another case. Repeat steps b-d using the collection date of the first
+                lab outside the 90 day window as the new earliest positive specimen
+                collection date.
+            ii. If not present then all covid cases for the patient have been
+                identified.
+    2.  If there are no positive labs than the patient has not been infected and no
+        cases are identified.
+
+    **** WARNING ****
+
+    As currently written this function applies an approximation of the official CDC/CTSE
+    COIVD case defintion. In its current form this function is intended as a proof of
+    concept for development purposes.
+
+    Known Limitations:
+    1.  Suspect cases resulting from antibody tests are not considered.
+    2.  No check has been implemented for a negative PCR within 48 hours of a positive
+        Antigen that would negate the case.
+    3.  No consideration has been given to how the case definition has changed over the
+        course of the pandemic.
+    """
+    case_list = []
+    patient_labs.sort_values(by=["effectiveDateTime"], ascending=True)
+    positive_labs = patient_labs.loc[patient_labs.result == "POSITIVE"].reset_index()
+    remaining_labs = len(positive_labs)
+    while remaining_labs > 0:
+        collection_date = positive_labs.effectiveDateTime[0]
+        current_infection_labs = positive_labs.loc[
+            positive_labs.effectiveDateTime.map(lambda x: (x - collection_date).days)
+            <= 90
+        ]
+        if "pcr" in current_infection_labs.type.to_list():
+            status = "confirmed"
+        else:
+            status = "probable"
+        case = patient_labs.agg(record_combination_func)
+        case["effectiveDateTime"] = collection_date
+        case["status"] = status
+        case_list.append(case)
+        positive_labs = positive_labs.loc[
+            positive_labs.effectiveDateTime.map(lambda x: (x - collection_date).days)
+            > 90
+        ].reset_index()
+        remaining_labs = len(positive_labs)
+    return case_list
+
+
+if __name__ == "__main__":
+
+    # Set values that specify a blob to load.
+    STORAGE_ACCOUNT_URL = "https://pitestdatasa.blob.core.windows.net"
+    CONTAINER_NAME = "bronze"
+    ELR_FILE_NAME = "csv-test/elr.csv"
+    COVID_CASE_DATAMART_FILENAME = "datamart-test/covid_case_datamart.csv"
+
+    # Load data.
+    labs = pd.read_csv(read_blob(STORAGE_ACCOUNT_URL, CONTAINER_NAME, ELR_FILE_NAME))
+    covid_loincs = pd.read_csv(
+        pathlib.Path(__file__).parent / "assets" / "covid_loinc_reference.csv"
+    )
+
+    # Clean and standardize data.
+    covid_labs = pd.merge(labs, covid_loincs, how="inner", on="loincCode")
+    covid_labs = covid_labs.loc[covid_labs.type.isin(["pcr", "ag"])]
+    covid_labs.drop(["loincCode", "notes"], axis=1, inplace=True)
```

Comment:
Just a fun tip, when you're invoking `.drop()` you can actually just give it a `columns` parameter instead of having to muck around with an axis, e.g. `covid_labs.drop(columns=['loincCode', 'notes'])`.